### PR TITLE
Finish show words

### PR DESCRIPTION
**Pull request by vietanhbkaaa**

### CODE REVIEW

File: app/assets/javascripts/words.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (phuong1492) [inline app/assets/javascripts/words.coffee]: remove file nay di

File: app/assets/stylesheets/words.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the words controller here.
```
Comment (phuong1492) [inline app/assets/stylesheets/words.scss]: remove file

File: app/views/lessons/show.html.erb [hunk 3]
```diff
@@ -16,4 +16,4 @@
     <% end %>
   <% end %>
   <%= f.submit "Submit", class: "btn btn-primary" %>
```
Comment (phuong1492) [inline app/views/lessons/show.html.erb]: I18n

File: app/views/static_pages/home.html.erb [hunk 4]
```diff
@@ -8,7 +8,7 @@
     </aside>
     <div class="col-md-8">
       <section class="center">
-        <%= link_to(t("links.word"),'#', class: "btn btn-lg btn-primary")%>
+        <%= link_to(t("links.word"), words_path, class: "btn btn-lg btn-primary")%>
         <%= link_to(t("links.course"), courses_path, class: "btn btn-lg btn-primary")%>
       </section>
       <h1>Activity Feed</h1>
```
Comment (phuong1492) [inline app/views/static_pages/home.html.erb]: I18n

File: app/views/static_pages/home.html.erb [hunk 5]
```diff
@@ -8,7 +8,7 @@
     </aside>
     <div class="col-md-8">
       <section class="center">
-        <%= link_to(t("links.word"),'#', class: "btn btn-lg btn-primary")%>
+        <%= link_to(t("links.word"), words_path, class: "btn btn-lg btn-primary")%>
```
Comment (phuong1492) [inline app/views/static_pages/home.html.erb]: remove ()

File: app/views/words/_show.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<div class="col-md-3 col-sm-offset-8">
+  <%= form_tag words_path, method: "get" do %>
+    <%= select_tag :search, options_for_select(Course.all.map{|course| course.name}, params[:search]) %>
+    <div class="radio-inline">
+      <%= radio_button_tag :stt, "all" %>
+      <%= label_tag(:stt_all, "All") %>
```
Comment (phuong1492) [inline app/views/words/_show.html.erb]: I18n

File: app/controllers/words_controller.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      @courses = current_user.courses
+      @lessons = @course1.lessons
+      @words = @course1.words.paginate page: params[:page], per_page: 10
+      if params[:stt]
+        option_ids = []
+        @lessons.each do |lesson|
+          lesson.answers.select{|ans| option_ids << ans.option_id if ans.option.correct?}
+        end
+        @answers = Answer.true_answer @lessons.ids, option_ids
+        word_ids = []
```
Comment (binhthanhnguyen) [inline app/controllers/words_controller.rb]: not use array like this. use Array.new

File: app/controllers/words_controller.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        @answers.each {|a| word_ids << a.word_id}
+        @words = @course1.words.send(params[:stt], word_ids).
+          paginate page: params[:page], per_page: 10
+      end
+    else
+      @courses = current_user.courses
+      @words =[]
```
Comment (binhthanhnguyen) [inline app/controllers/words_controller.rb]: Array.new

File: app/views/static_pages/home.html.erb [hunk 9]
```diff
@@ -11,8 +11,8 @@
     </aside>
     <div class="col-md-8">
       <section class="center">
-        <%= link_to(t("links.word"),'#', class: "btn btn-lg btn-primary")%>
-        <%= link_to(t("links.course"), courses_path, class: "btn btn-lg btn-primary")%>
+        <%= link_to t("links.word"), user_words_path(@user), class: "btn btn-lg btn-primary" %>
```
Comment (phuong1492) [inline app/views/static_pages/home.html.erb]: remove ()

File: app/views/words/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<% provide :title, "All words" %>
+<h1> All words</h1>
+<div class="row">
+  <div class="col-xs-12 ">
+    <%= render "show" %><br>
+  </div>
+  <div class="col-xs-12">
+    <ul class="words">
+    <%= will_paginate @words %>
```
Comment (phuong1492) [inline app/views/words/index.html.erb]: ident

File: app/controllers/words_controller.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+      @courses.each do |course|
+        course.words.each{|word| @words.push(word)}
```
Comment (phuong1492) [inline app/controllers/words_controller.rb]: can remove ()?

File: app/controllers/words_controller.rb [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      @words = @course1.words.paginate page: params[:page], per_page: 10
+      if params[:stt]
+        option_ids = []
+        @lessons.each do |lesson|
+          lesson.answers.select{|ans| option_ids << ans.option_id if ans.option.correct?}
+        end
+        @answers = Answer.true_answer @lessons.ids, option_ids
+        word_ids = []
+        @answers.each {|a| word_ids << a.word_id}
+        @words = @course1.words.send(params[:stt], word_ids).
```
Comment (phuong1492) [inline app/controllers/words_controller.rb]: Can remove()?

File: app/controllers/words_controller.rb [hunk 13]
```diff
@@ -0,0 +1,29 @@
+class WordsController < ApplicationController
+  require "will_paginate/array"
+  def index
+    if params[:search]
+      @course1 = Course.find_by(name: params[:search])
```
Comment (phuong1492) [inline app/controllers/words_controller.rb]: can remove ()?

File: app/models/word.rb [hunk 14]
```diff
@@ -8,7 +8,12 @@ class Word < ActiveRecord::Base
 
   validates :content, presence: true
 
+  scope :learned, ->user{where id: Answer.select(:word_id ).where("lesson_id  IN (?) and option_id is not null", user.lessons.ids)}
```
Comment (ThuBM) [inline app/models/word.rb]: split to 2 rows

File: app/views/words/index.html.erb [hunk 15]
```diff
@@ -0,0 +1,13 @@
+<% provide :title, "All words" %>
+<h1> All words</h1>
+<div class="row">
+  <div class="col-xs-12 ">
+    <%= render "show" %><br>
+  </div>
+  <div class="col-xs-12">
+    <ul class="words">
+      <%= will_paginate @words %>
```
Comment (ThuBM) [inline app/views/words/index.html.erb]: remove @words

File: config/routes.rb [hunk 16]
```diff
@@ -9,9 +9,10 @@
   post "login" => "sessions#create"
   delete "logout" => "sessions#destroy"
 
+
   resources :users, except: :destroy do
-    match "/:relationship" => "relationships#index", as: :relationship,
-      via: :get, constraints: {relationship: /(following|followers)/}
+    resources :words
+    match "/:relationship" => "relationships#index", as: :relationship, via: :get, constraints: {relationship: /(following|followers)/}
```
Comment (ThuBM) [inline config/routes.rb]: split to 2 rows

File: app/models/word.rb [hunk 17]
```diff
@@ -8,7 +8,14 @@ class Word < ActiveRecord::Base
 
   validates :content, presence: true
 
+  scope :learned, ->user{where id: Answer.select(:word_id ).
+    where("lesson_id  IN (?) and option_id is not null", user.lessons.ids)}
+  scope :not_learned, ->user{where id: Answer.select(:word_id).
+    where(lesson_id: user.lessons.ids, option: nil)}
+  scope :get_all, ->user{where id: Answer.select(:word_id).where(lesson_id: user.lessons.ids)}
```
Comment (dieunb) [inline app/models/word.rb]: Sub query

File: app/models/word.rb [hunk 18]
```diff
@@ -8,7 +8,14 @@ class Word < ActiveRecord::Base
 
   validates :content, presence: true
 
+  scope :learned, ->user{where id: Answer.select(:word_id ).
+    where("lesson_id  IN (?) and option_id is not null", user.lessons.ids)}
+  scope :not_learned, ->user{where id: Answer.select(:word_id).
```
Comment (dieunb) [inline app/models/word.rb]: Sub query

File: app/models/word.rb [hunk 19]
```diff
@@ -8,7 +8,14 @@ class Word < ActiveRecord::Base
 
   validates :content, presence: true
 
+  scope :learned, ->user{where id: Answer.select(:word_id ).
```
Comment (dieunb) [inline app/models/word.rb]: Sub query

File: app/models/word.rb [hunk 20]
```diff
@@ -8,7 +8,17 @@ class Word < ActiveRecord::Base
 
   validates :content, presence: true
 
+  scope :get_all, ->user{
```
Comment (dieunb) [inline app/models/word.rb]: `scope :get_all, ->user{}`

**Pull request merged**